opmon/monitoring.py

```diff
@@ -89,28 +92,15 @@ def _run_metrics_sql(self, submission_date: datetime):
             print(f"Failed to run opmon project: {e}")
             return
 
-        try:
-            self.bigquery.client.get_table(
-                f"{self.derived_dataset}.{self.normalized_slug}_v{SCHEMA_VERSIONS['metric']}"
-            )
-            date_partition = str(submission_date).replace("-", "").split(" ")[0]
-            destination_table = (
-                f"{self.normalized_slug}_v{SCHEMA_VERSIONS['metric']}${date_partition}"
-            )
-            self.bigquery.execute(
-                self._get_metrics_sql(submission_date=submission_date),
-                destination_table,
-                write_disposition=bigquery.job.WriteDisposition.WRITE_TRUNCATE,
-                dataset=f"{self.derived_dataset}",
-            )
-        except cloud.exceptions.NotFound:
-            self.bigquery.execute(
-                self._get_metrics_sql(submission_date=submission_date),
-                f"{self.normalized_slug}_v{SCHEMA_VERSIONS['metric']}",
-                clustering=["build_id"],
-                time_partitioning="submission_date",
-                dataset=f"{self.derived_dataset}",
-            )
+        table_name = f"{self.normalized_slug}_v{SCHEMA_VERSIONS['metric']}"
+        self.bigquery.execute(
+            self._get_metrics_sql(submission_date=submission_date, table_name=table_name),
+            destination=f"{table_name}${submission_date:%Y%m%d}",
+            clustering=["build_id"],
+            time_partitioning="submission_date",
+            write_disposition=bigquery.job.WriteDisposition.WRITE_TRUNCATE,
+            dataset=self.derived_dataset,
+        )
```

Comment:
by specifying clustering, time partitioning, and a partition id, BigQuery will create the table as needed, instead of repeating logic based on whether or not the destination table exists